google/cloud/storage/tests/object_integration_test.cc

```diff
@@ -127,6 +132,43 @@ TEST_F(ObjectIntegrationTest, BasicReadWrite) {
   client.DeleteObject(bucket_name, object_name);
 }
 
+TEST_F(ObjectIntegrationTest, AccessControlCRUD) {
+  gcs::Client client;
+  auto bucket_name = ObjectTestEnvironment::bucket_name();
+  auto object_name = MakeRandomObjectName();
+
+  // Create the object, but only if it does not exist already.
+  (void)client.InsertObject(bucket_name, object_name, LoremIpsum(),
+                            gcs::IfGenerationMatch(0));
+
+  auto entity_name = MakeEntityName();
+  std::vector<gcs::ObjectAccessControl> initial_acl =
+      client.ListObjectAcl(bucket_name, object_name);
+
+  auto name_counter = [](std::string const& name,
+                         std::vector<gcs::ObjectAccessControl> const& list) {
+    auto name_matcher = [](std::string const& name) {
+      return [name](gcs::ObjectAccessControl const& m) {
+        return m.entity() == name;
+      };
+    };
+    return std::count_if(list.begin(), list.end(), name_matcher(name));
+  };
+  ASSERT_EQ(0, name_counter(entity_name, initial_acl))
+      << "Test aborted. The entity <" << entity_name << "> already exists."
+      << "This is unexpected as the test generates a random object name.";
+
+  auto result =
```

Comment:
Instead of `auto`, prefer explicitly stating the type here, since it's not stated anywhere in the surrounding context and you call a method on `result` a few lines down.

Reply:
Fixed.